agent/build.gradle

```diff
@@ -28,33 +36,44 @@ CopySpec isolateSpec() {
     }
 }
 
+// includes everything except otel agent
+task customShadow(type: ShadowJar) {
+    from sourceSets.main.output
+
+    configurations = [project.configurations.customShadowInclude]
+
+    dependsOn ':custom:shadowJar'
+    dependsOn ':instrumentation:shadowJar'
+    dependsOn ':bootstrap:jar'
+    with isolateSpec()
+
+    mergeServiceFiles()
+    exclude("**/module-info.class")
+
+    // Prevents conflict with other SLF4J instances. Important for premain.
+    relocate("org.slf4j", "io.opentelemetry.javaagent.slf4j")
+    // rewrite dependencies calling Logger.getLogger
+    relocate("java.util.logging.Logger", "io.opentelemetry.javaagent.bootstrap.PatchLogger")
+
+    // prevents conflict with library instrumentation
+    relocate("io.opentelemetry.instrumentation.api", "io.opentelemetry.javaagent.shaded.instrumentation.api")
+
+    // relocate OpenTelemetry API
+    relocate("io.opentelemetry.api", "io.opentelemetry.javaagent.shaded.io.opentelemetry.api")
+    relocate("io.opentelemetry.spi", "io.opentelemetry.javaagent.shaded.io.opentelemetry.spi")
+    relocate("io.opentelemetry.context", "io.opentelemetry.javaagent.shaded.io.opentelemetry.context")
+}
+
 tasks {
+    // merge otel agent with our agent
     shadowJar {
-        dependsOn ':custom:shadowJar'
-        dependsOn ':instrumentation:shadowJar'
-        dependsOn ':bootstrap:jar'
-        with isolateSpec()
+        from tasks.customShadow.outputs
 
-        /// TODO - with this set to EXCLUDE, service files will not be merged between instrumentation and custom
-        duplicatesStrategy = DuplicatesStrategy.INCLUDE
```

Comment:
Is this TODO now solved?

Reply:
Yes. I compared extracted jar before and after changes and it had the following differences.
- com/splunk/opentelemetry/SplunkAgent.class has constant pool elements in different order, otherwise classes are identical. New version matches the one in agent/build/classes
- ordering of elements in inst/META-INF/services files is different. Previously entries from splunk were before otel now they are after.
- new jar does not have a directory entry for inst/META-INF/services, idk where it went, maybe mergeServiceFiles does something with it. But it shouldn't be needed anyway, directory entries in jar files are optional and as far as I understand won't affect service loader.

Reply:
The problem with previous TODO was that if there were service files in both upstream and custom, then they *must* be merged, but with `EXCLUDE` Strategy instead of merging just one was randomly kept.

Reply:
verified, service files are merged

Reply:
verified, service files are merged